[Swiftify] Don't create safe wrapper for autoreleasing pointers

hnrklssn opened the pull request:
_SwiftifyImport doesn't know how to handle AutoreleasingUnsafeMutablePointer, so we should not attach any .countedBy information for pointers that are imported as this type.

This also adds defensive checks against adding .countedBy to any pointer type that _SwiftifyImport doesn't know how to transform.

rdar://151479521

hnrklssn committed:
[Swiftify] Don't create safe wrapper for autoreleasing pointers

_SwiftifyImport doesn't know how to handle
AutoreleasingUnsafeMutablePointer, so we should not attach any
.countedBy information for pointers that are imported as this type.

This also adds defensive checks against adding .countedBy to any pointer
type that _SwiftifyImport doesn't know how to transform.

rdar://151479521

hnrklssn commented:
@swift-ci please smoke test

j-hui commented on lib/ClangImporter/ImportDecl.cpp:
```diff
@@ -9192,6 +9192,14 @@ static bool SwiftifiableCAT(const clang::CountAttributedType *CAT) {
+  Type nonnullType = swiftType->lookThroughSingleOptionalType();
+  PointerTypeKind PTK;
+  return nonnullType->isOpaquePointer() ||
+    (nonnullType->getAnyPointerElementType(PTK) && PTK != PTK_AutoreleasingUnsafeMutablePointer);
```
Since you are already checking for and allowing types imported as opaque pointers here, can you add some test cases to make sure those work?

hnrklssn replied:
We have a test case of that here: https://github.com/swiftlang/swift/blob/f57e2d7676e18f513afb219dbc1c2c738af4e019/test/Interop/C/swiftify-import/Inputs/sized-by.h#L23

j-hui reviewed (approved):
LGTM, aside from a testing suggestion

hnrklssn committed:
[Swiftify] Move interop test case to ObjC directory

This test case fails on platforms without Objective-C interop. Move it
to the ObjC directory, and add a local config that disables the entire
subdirectory if Objective-C interop is missing.

hnrklssn commented:
@swift-ci please smoke test

hnrklssn commented:
@swift-ci please smoke test linux

hnrklssn commented:
@swift-ci please smoke test linux
